ecs-cli/modules/clients/aws/servicediscovery/client_test.go

```diff
@@ -0,0 +1,214 @@
+			},
+		},
+	}
+}
+
+func setupMocksForTestFindPrivateNamespace(t *testing.T) {
+	// Mock Data
+	var namespaceData = map[string]servicediscovery.Namespace{
+		"namespace1": servicediscovery.Namespace{
+			Id:   aws.String("namespace1"),
+			Name: aws.String("corp"),
+			Properties: &servicediscovery.NamespaceProperties{
+				DnsProperties: &servicediscovery.DnsProperties{
+					HostedZoneId: aws.String("zone1"),
+				},
+			},
+		},
+		"namespace2": servicediscovery.Namespace{
+			Id:   aws.String("namespace2"),
+			Name: aws.String("prod"),
+			Properties: &servicediscovery.NamespaceProperties{
+				DnsProperties: &servicediscovery.DnsProperties{
+					HostedZoneId: aws.String("zone2"),
+				},
+			},
+		},
+		"namespace3": servicediscovery.Namespace{
+			Id:   aws.String("namespace3"),
+			Name: aws.String("corp"),
+			Properties: &servicediscovery.NamespaceProperties{
+				DnsProperties: &servicediscovery.DnsProperties{
+					HostedZoneId: aws.String("zone3"),
+				},
+			},
+		},
+		"namespace4": servicediscovery.Namespace{
+			Id:   aws.String("namespace4"),
+			Name: aws.String("bridge"),
+			Properties: &servicediscovery.NamespaceProperties{
+				DnsProperties: &servicediscovery.DnsProperties{
+					HostedZoneId: aws.String("zone4"),
+				},
+			},
+		},
+		"namespace5": servicediscovery.Namespace{
+			Id:   aws.String("namespace5"),
+			Name: aws.String("bridge"),
+			Properties: &servicediscovery.NamespaceProperties{
+				DnsProperties: &servicediscovery.DnsProperties{
+					HostedZoneId: aws.String("zone5"),
+				},
+			},
+		},
+	}
+
+	var hostedZoneData = map[string]route53.GetHostedZoneOutput{
+		"zone1": route53.GetHostedZoneOutput{
+			VPCs: []*route53.VPC{
+				&route53.VPC{
+					VPCId:     aws.String("vpc-8BAADF00D"),
+					VPCRegion: aws.String("us-east-1"),
+				},
+			},
+		},
+		"zone2": route53.GetHostedZoneOutput{
+			VPCs: []*route53.VPC{
+				&route53.VPC{
+					VPCId:     aws.String("vpc-1CEB00DA"),
+					VPCRegion: aws.String("us-east-1"),
+				},
+			},
+		},
+		"zone3": route53.GetHostedZoneOutput{
+			VPCs: []*route53.VPC{
+				&route53.VPC{
+					VPCId:     aws.String("vpc-C00010FF"),
+					VPCRegion: aws.String("sa-east-1"),
+				},
+			},
+		},
+		"zone4": route53.GetHostedZoneOutput{
+			VPCs: []*route53.VPC{
+				&route53.VPC{
+					VPCId:     aws.String("vpc-C00010FF"),
+					VPCRegion: aws.String("ap-south-1"),
+				},
+				&route53.VPC{
+					VPCId:     aws.String("vpc-D15EA5E"),
+					VPCRegion: aws.String("ap-south-1"),
+				},
+			},
+		},
+		"zone5": route53.GetHostedZoneOutput{
+			VPCs: []*route53.VPC{
+				&route53.VPC{
+					VPCId:     aws.String("vpc-DEADBAAD"),
+					VPCRegion: aws.String("ap-south-1"),
+				},
+				&route53.VPC{
+					VPCId:     aws.String("vpc-D15EA5E"),
+					VPCRegion: aws.String("ap-south-1"),
+				},
+			},
+		},
+	}
+
+	// Mock API Actions
+	listNamespaces = func(isPublic bool, params *config.CommandConfig, fn func(*servicediscovery.NamespaceSummary) bool) error {
```

Comment:
Again, I like that we're trying to avoid too much clunky mocking in the tests, but I'm wondering if doing it this way makes the test a bit to knowledgable/coupled with the implementation.